File: app/src/main/java/com/mesada/shop/model/Product.java
package com.mesada.shop.model;

import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.databinding.BindingAdapter;
import androidx.recyclerview.widget.DiffUtil;

import com.bumptech.glide.Glide;
import com.mesada.shop.R;

import java.util.Objects;

public class Product {
    private String productid;
    private String productName;
    private String productCategory;
    private double productPrice;
    private String bgColor;

    public Product(String productid, String productName, String productCategory, double productPrice, String bgColor) {
        this.productid = productid;
        this.productName = productName;
        this.productCategory = productCategory;
        this.productPrice = productPrice;
        this.bgColor = bgColor;
    }

    public String getId() {
        return productid;
    }

    public void setId(String productid) {
        this.productid = productid;
    }

    public String getName() {
        return productName;
    }

    public void setName(String productName) {
        this.productName = productName;
    }

    public double getPrice() {
        return productPrice;
    }

    public void setPrice(double productPrice) {
        this.productPrice = productPrice;
    }

    public String getCategory() {
        return productCategory;
    }

    public void setCategory(String productCategory) {
        this.productCategory = productCategory;
    }

    public String getBgColor() {
        return bgColor;
    }

    public void setBgColor(String bgColor) {
        this.bgColor = bgColor;
    }

    public int getImageUrl() {
        {
            switch (productid)
            {
                case "p_1":
                    return R.drawable.p_1;
                case "p_2":
                    return R.drawable.p_2;
                case "p_3":
                    return R.drawable.p_3;
                case "p_4":
                    return R.drawable.p_4;
                case "p_5":
                    return R.drawable.p_5;
            }
        }
        return R.drawable.ic_launcher_background;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Product product = (Product) o;
        return Double.compare(product.getPrice(), getPrice()) == 0 &&
                getCategory().equals(product.getCategory()) &&
                getId().equals(product.getId()) &&
                getName().equals(product.getName()) &&
                getBgColor().equals(product.getBgColor());
    }

    public static DiffUtil.ItemCallback<Product> itemCallback = new DiffUtil.ItemCallback<Product>() {
        @Override
        public boolean areItemsTheSame(@NonNull Product oldItem, @NonNull Product newItem) {
            return oldItem.getId().equals(newItem.getId());
        }

        @Override
        public boolean areContentsTheSame(@NonNull Product oldItem, @NonNull Product newItem) {
            return oldItem.equals(newItem);
        }
    };


    @BindingAdapter("android:productImage")
    public static void loadImage(ImageView imageView, String imageUrl) {
        Glide.with(imageView)
                .load(imageUrl)
                .fitCenter()
                .into(imageView);
    }
}

File: app/src/main/java/com/mesada/shop/model/Product.kt
package com.mesada.shop.model

import android.util.Log
import android.widget.ImageView
import androidx.recyclerview.widget.DiffUtil
import com.bumptech.glide.Glide
import com.mesada.shop.R

class Product(var id: String, var name: String, var category: String, var price: Double, var bgColor: String) {

    val imageUrl: Int
        get() {
            run {
                when (this.id) {
                    "p_1" -> return R.drawable.pic_1
                    "p_2" -> return R.drawable.pic_2
                    "p_3" -> return R.drawable.pic_3
                    "p_4" -> return R.drawable.pic_4
                    "p_5" -> return R.drawable.pic_5
                    else -> return R.drawable.pic_5
                }
            }
            return R.drawable.ic_launcher_background
        }

    val imageResourceBg: Int
        get() {
            run {
                when (id) {
                    "p_1" -> return R.drawable.p_1
                    "p_2" -> return R.drawable.p_2
                    "p_3" -> return R.drawable.p_3
                    "p_4" -> return R.drawable.p_1
                    "p_5" -> return R.drawable.p_2
                    else -> return R.drawable.p_3
                }
            }
            return R.drawable.ic_launcher_background
        }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val product = o as Product
        return java.lang.Double.compare(product.price, price) == 0 && category == product.category && id == product.id && name == product.name && bgColor == product.bgColor
    }

    companion object {
        @JvmField
        var itemCallback: DiffUtil.ItemCallback<Product> = object : DiffUtil.ItemCallback<Product>() {
            override fun areItemsTheSame(oldItem: Product, newItem: Product): Boolean {
                return oldItem.id == newItem.id
            }

            override fun areContentsTheSame(oldItem: Product, newItem: Product): Boolean {
                return oldItem == newItem
            }
        }

        fun loadImage(imageView: ImageView?, imageUrl: String?) {
            Glide.with(imageView!!)
                    .load(imageUrl)
                    .fitCenter()
                    .into(imageView)
        }
    }

}